File: app/src/test/java/com/annushkaproject/programmerscalculator/StandardTests/StandardMultiplicationTest.java
package com.annushkaproject.programmerscalculator.StandardTests;

import com.annushkaproject.programmerscalculator.model.CalculationModel;
import com.annushkaproject.programmerscalculator.model.Value;
import com.annushkaproject.programmerscalculator.utils.StandardOperationsUtil;

import org.junit.Test;

import static com.annushkaproject.programmerscalculator.model.Operator.MULTIPLY;
import static org.junit.Assert.assertEquals;

public class StandardMultiplicationTest {
    @Test
    public void standardMultiplication_isCorrect() {
        assertEquals(25, StandardOperationsUtil.calculateResultForTwoSidedOperator(new CalculationModel(new Value(5), new Value(5), MULTIPLY)), 0); // Integer test
        assertEquals(-25, StandardOperationsUtil.calculateResultForTwoSidedOperator(new CalculationModel(new Value(-5), new Value(5), MULTIPLY)), 0); // Integer test
        assertEquals(0.00001, StandardOperationsUtil.calculateResultForTwoSidedOperator(new CalculationModel(new Value(0.000005), new Value(2), MULTIPLY)), 0); // Decimal test
        assertEquals(3, StandardOperationsUtil.calculateResultForTwoSidedOperator(new CalculationModel(new Value(1.5), new Value(2), MULTIPLY)), 0); // Decimal test
        assertEquals(1, StandardOperationsUtil.calculateResultForTwoSidedOperator(new CalculationModel(new Value(2), new Value(0.5), MULTIPLY)), 0); // Decimal test
    }
}

File: app/src/test/java/com/annushkaproject/programmerscalculator/StandardTests/StandardMultiplicationTest.kt
package com.annushkaproject.programmerscalculator.StandardTests

import com.annushkaproject.programmerscalculator.model.CalculationModel
import com.annushkaproject.programmerscalculator.model.Operator
import com.annushkaproject.programmerscalculator.utils.StandardOperationsUtil.calculateResultForTwoSidedOperator
import org.junit.Assert
import org.junit.Test
import java.math.BigDecimal

class StandardMultiplicationTest {
    @Test
    fun testPositiveIntegers() {
        Assert.assertEquals(25.0, calculateResultForTwoSidedOperator(CalculationModel(BigDecimal.valueOf(5), BigDecimal.valueOf(5), Operator.MULTIPLY)), 0.0)
    }

    @Test
    fun testPositiveAndNegativeIntegers() {
        Assert.assertEquals(-25.0, calculateResultForTwoSidedOperator(CalculationModel(BigDecimal.valueOf(-5), BigDecimal.valueOf(5), Operator.MULTIPLY)), 0.0)
    }

    @Test
    fun testDecimalSmall() {
        Assert.assertEquals(0.00001, calculateResultForTwoSidedOperator(CalculationModel(BigDecimal.valueOf(0.000005), BigDecimal.valueOf(2), Operator.MULTIPLY)), 0.0)
    }

    @Test
    fun testFirstDecimalParameter() {
        Assert.assertEquals(3.0, calculateResultForTwoSidedOperator(CalculationModel(BigDecimal.valueOf(1.5), BigDecimal.valueOf(2), Operator.MULTIPLY)), 0.0)
    }

    @Test
    fun testSecondDecimalParameter() {
        Assert.assertEquals(1.0, calculateResultForTwoSidedOperator(CalculationModel(BigDecimal.valueOf(2), BigDecimal.valueOf(0.5), Operator.MULTIPLY)), 0.0)
    }
}